samcli/local/layers/layer_downloader.py

```diff
@@ -91,7 +97,18 @@ def download(self, layer, force=False):
         """
         if layer.is_defined_within_template:
             LOG.info("%s is a local Layer in the template", layer.name)
-            layer.codeuri = resolve_code_path(self.cwd, layer.codeuri)
+            # the template file containing the layer might not be in the same directory as root template file
+            # therefore we need to join the path of template directory and codeuri in case codeuri is a relative path.
+            stacks = [stack for stack in self._stacks if stack.stack_path == layer.stack_path]
+            if not stacks:
+                raise RuntimeError(f"Cannot find stack that matches layer's stack_path {layer.stack_path}")
+            stack = stacks[0]
```

Comment:
we are sure there is at most one matching stack (no two stacks have the same stack path), so having a list of stacks is confusing, use the `next` method instead. `stack = next(stack for stack in self._stacks if stack.stack_path == layer.stack_path)`

Reply:
Should know `next()` earlier, thx, changed.